src/services/SessionService.js

```diff
@@ -8,20 +8,24 @@ export default {
   loading: false,
   currentSession: {
     sessionId: null,
-    data: {},
+    data: {
+      volunteerJoinTime: null,
+      sessionEndTime: null,
+    },
   },
 
   getPartner() {
     const user = UserService.getUser();
     const session = this.currentSession.data;
-
     if (user.isVolunteer) {
+      this.currentSession.data.volunteerJoinTime = new Date();
       return session.student;
     }
     return session.volunteer;
   },
 
   endSession(options = {}) {
```

Comment:
@joncancode I didn't find any part of the code that deals with storing the data in `currentSession.data` into the DB. This is necessary in order to consider this feature complete.

You'll also need to modify the backend so that it can handle the data that you're sending about the session.